storage_controller/src/service.rs

```diff
@@ -4312,6 +4312,13 @@ impl Service {
                         continue;
                     }
 
+                    if !new_nodes.values().any(Node::is_available) {
+                        // Special case for when all nodes are unavailable: there is no point
```

Comment:
Should this be `Node::may_schedule`?  We want to know "when we schedule, will be find something that we can attach to"?

Also, perhaps we can pull this check out of the loop over tenant shards: may_schedule isn't particularly expensive, but it'll give the same answer every time in this loop.

Reply:
`may_schedule` would also work fine, but I have a preference for the more restrictive `is_available`. Skimming the cancellation, heartbeat handling (hence triggering the cancellation token) shouldn't overlap with this check, but there's nothing preventing it in theory.

Reply:
I'm thinking of may_schedule as the more restrictive one: the case where we have some nodes with scheduling mode Paused or similar, it'll pass the is_available check.  But it's kind of an edge case: if all-but-one of the nodes is down and the one that's left is Paused.

Reply:
Okay, I'm convinced. Thanks for bearing with: https://github.com/neondatabase/neon/pull/8060/commits/0ec3bc2135de89bbf0bfe25e2f66f36cf808fbfe